### PR TITLE
Merge the new vbox plugin into the existing virtualbox plugin

### PR DESCRIPTION
**Pull request by tas50**
It makes sense to handle both guest and host logic in the same plugin since you can't be both. This gives a more predictable namespace and makes the code slightly smaller. I also removed the ostypes field since that doesn't change by system and is an enormous amount of data to collect with pretty limited value from a cookbook standpoint.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/ohai/plugins/virtualbox.rb [hunk 1]
```diff
@@ -20,8 +22,128 @@
...
+        left, right = line.split("=")
+
+        # remove enclosing quotes, if needed
+        key =
+          case left
+          when /^"(.*)"$/
+            Regexp.last_match(1)
+          else
+            left
+          end
```
Comment (lamont-granquist) [inline lib/ohai/plugins/virtualbox.rb]: probably:

```
key = left
key = Regexp.last_match(1) if left ~= /^"(.*)"$/
```

Comment (tas50) [reply lib/ohai/plugins/virtualbox.rb]: Updated using delete_prefix / delete_suffix per our review

File: lib/ohai/plugins/virtualbox.rb [hunk 2]
```diff
@@ -20,8 +22,113 @@
...
+    if so.exitstatus == 0
+      so.stdout.lines.each do |line|
+        line.chomp!
+        left, right = line.split("=")
+
+        # remove enclosing quotes, if needed
+        key = left.delete_prefix('"').delete_suffix('"')
+
+        # skip the name attribute since that is the parent key
+        next if left == "name"
```
Comment (robbkidd) [inline lib/ohai/plugins/virtualbox.rb]: Maybe want to test `key` here after the `"` chomping on `left`. Is the value always lowercase `name` or should this test be after downcasing?

Comment (tas50) [reply lib/ohai/plugins/virtualbox.rb]: Updated

**Pull request merged**